libyul/AST.h

```diff
@@ -69,14 +73,18 @@ class LiteralValue {
 struct Literal { langutil::DebugData::ConstPtr debugData; LiteralKind kind; LiteralValue value; };
 /// External / internal identifier or label reference
 struct Identifier { langutil::DebugData::ConstPtr debugData; YulName name; };
+/// Builtin function
+struct BuiltinName { langutil::DebugData::ConstPtr debugData; BuiltinHandle handle; };
+/// Identifier for function names
+using FunctionHandle = std::variant<YulName, BuiltinHandle>;
```

Comment:
We could add a bit more detail here:
```suggestion
/// AST Node representing a reference to one of the built-in functions (as defined by the dialect).
/// In the source it's an actual name, while in the AST we only store a handle that can be used to find the the function in the Dialect.
struct BuiltinName { langutil::DebugData::ConstPtr debugData; BuiltinHandle handle; };
/// Type that can refer to both user-defined functions and built-ins.
/// Technically the AST allows these names to overlap, but this is not possible to represent in the source.
using FunctionHandle = std::variant<YulName, BuiltinHandle>;
```